### PR TITLE
Seeing friend's profile feature has been implemented

### LINKED ISSUE #164: Open a user's profile page when someone clicks on his name in my trips
In my trips, if someone clicks on any user's name, open that user's profile page similar to what we see in my profile currently 

+ On clicking the + icon, the view should show/hide
+ Show the arrow icon on each friend item
+ On clicking anywhere on the list item, the user's profile page should open up.

### PR DESCRIPTION
**Pull request by aayushinigam**
# Description
User can now see his/her friend's profile.
ss1
ss2
ss3

Fixes #164 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
Successfully ran on an android device.
Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
Use vector assets instead of png icons.

**Comment by Swati4star** [comment]
Also verify this when there are no friends in a trip. That UI should not get affected by this.

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+        mHandler = new Handler(Looper.getMainLooper());
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+        mToken = mSharedPreferences.getString(USER_TOKEN, null);
+
+        getFriendDetails(String.valueOf(friendId));
+
+        Objects.requireNonNull(getSupportActionBar()).setHomeButtonEnabled(true);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java]: Remove these extra lines

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    public static Intent getStartIntent(Context context, int id) {
+        Intent intent = new Intent(context, FriendsProfileActivity.class);
+        intent.putExtra(EXTRA_MESSAGE_FRIEND_ID, id);
+        return intent;
+    }
+
+    private void getFriendDetails(final String friendId) {
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java]: Remove line

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                        String userName = object.getString("username");
+                        String firstName = object.getString("first_name");
+                        String lastName = object.getString("last_name");
+                        String fullname = firstName + " " + lastName;
+                        String imageURL = object.getString("image");
+                        String dateJoined = object.getString("date_joined");
+                        Long dateTime = rfc3339ToMills(dateJoined);
+                        String date = getDate(dateTime);
+                        friendUserName.setText(fullname);
+                        friendJoiningDate.setText("Joined on" + " " + date);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java]: Move this string to strings.xml

File: Android/app/src/main/res/layout/trip_friend_listitem.xml [hunk 4]
```diff
@@ -32,5 +30,16 @@
...
+    <ImageView
+        android:id="@+id/next_arrow"
+        android:layout_width="37dp"
+        android:layout_height="43dp"
+        android:background="@drawable/right_arrow_icon"
+        android:maxHeight="2dp"
+        android:maxWidth="2dp"
+        android:scaleType="fitEnd"
+        android:contentDescription="next arrow"/>
+
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/trip_friend_listitem.xml]: Remove these 2 extra lines

**Comment by Swati4star** [comment]
@aayushinigam We now have user status also. Incorporate that also in this screen.
Reference : https://github.com/project-travel-mate/Travel-Mate/pull/227

**Comment by aayushinigam** [comment]
@Swati4star Travis build fails saying "The command "./gradlew assembleDebug assembleRelease" exited with 1." 
Although the same command runs successfully on my PC.

**Comment by Swati4star** [comment]
See line 2812. The error message is displayed there. 
Did it get messed up while merging the code from master

**Comment by aayushinigam** [comment]
Ohh I get what's happening. Thanks I'll resolve it.

**Comment by Swati4star** [comment]
@aayushinigam Some of the bugs:
- When there are no friends, there shouldn't be this icon
screen shot 2018-07-09 at 11 02 07 pm

- When, I click on any trip, the UI is like this. This view should be hidden
screen shot 2018-07-09 at 11 03 30 pm

- When I click on plus icon, I get this error:
```
9 23:04:38.043 4980-4980/io.github.project_travel_mate E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.github.project_travel_mate, PID: 4980
    java.lang.NoClassDefFoundError: Failed resolution of: Lio/github/project_travel_mate/mytrips/MyTripFriendNameAdapter;
        at io.github.project_travel_mate.mytrips.MyTripInfoActivity$1$1.onClick(MyTripInfoActivity.java:211)
        at android.view.View.performClick(View.java:6256)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "io.github.project_travel_mate.mytrips.MyTripFriendNameAdapter" on path: DexPathList[[zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/base.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_dependencies_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_resources_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_0_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_1_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_2_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_3_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_4_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_5_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_6_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_7_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_8_apk.apk", zip file "/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_slice_9_apk.apk"],nativeLibraryDirectories=[/data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/lib/x86, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at io.github.project_travel_mate.mytrips.MyTripInfoActivity$1$1.onClick(MyTripInfoActivity.java:211) 
        at android.view.View.performClick(View.java:6256) 
        at android.view.View$PerformClick.run(View.java:24701) 
        at android.os.Handler.handleCallback(Handler.java:789) 
        at android.os.Handler.dispatchMessage(Handler.java:98) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6541) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
    	Suppressed: java.io.IOException: No original dex files found for dex location /data/app/io.github.project_travel_mate-kR73o9j2p7vsvSst28wghQ==/split_lib_resources_apk.apk
        at dalvik.system.DexFile.openDexFileNative(Native Method)
        at dalvik.system.DexFile.openDexFile(DexFile.java:353)
        at dalvik.system.DexFile.<init>(DexFile.java:100)
        at dalvik.system.DexFile.<init>(DexFile.java:74)
        at dalvik.system.DexPathList.loadDexFile(DexPathList.java:374)
        at dalvik.system.DexPathList.makeDexElements(DexPathList.java:337)
        at dalvik.system.DexPathList.<init>(DexPathList.java:157)
        at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:65)
        at dalvik.system.PathClassLoader.<init>(PathClassLoader.java:64)
        at com.android.internal.os.PathClassLoaderFactory.createClassLoader(PathClassLoaderFactory.java:43)
        at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:69)
        at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:36)
        at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:676)
        at android.app.LoadedApk.getClassLoader(LoadedApk.java:709)
        at android.app.LoadedApk.getResources(LoadedApk.java:936)
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2242)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5672)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
        at android.os.Handler.dispatchMessage(Handler.java:105)
```

**Comment by aayushinigam** [comment]
1. I'll do the modification required for 1st one.
2. I don't get it, which view should be hidden?
3. Errors is because there's no class named "MyTripFriendNameAdapter" but this class does exist in my PR. Are you sure you pulled the changes right?

**Comment by Swati4star** [comment]
- Friends in this trip & `+` icon should be hidden. They should up only after we have received the result from the API
- and yes, I have fetched your PR only. It has all the changes. I was facing that error. It's fine now. Ignore that error.
- Also, move the icons to the right of the screen. there's too much margin left on the right
screen shot 2018-07-09 at 11 25 35 pm
- If there's no status, hide the textview & icon. Don't show `No Status`
- The default strings doesn't look good:
screen shot 2018-07-09 at 11 26 50 pm
Add `--` as default value to every string

**Comment by Swati4star** [comment]
..and show a loader while you are fetching user's profile.

**Pull request merged**